gdal/port/cpl_error.cpp

```diff
@@ -193,6 +207,70 @@ void* CPL_STDCALL CPLGetErrorHandlerUserData(void)
         psCtx->psHandlerStack->pUserData : pErrorHandlerUserData );
 }
 
+
+void ApplyErrorHandler( CPLErrorContext *psCtx, CPLErr eErrClass,
+                        CPLErrorNum err_no, const char *pszMessage,
+                        bool is_cpldebug )
```

Comment:
is_cpldebug doesn't seem necessary. You could just test eErrClass == CE_Debug

Reply:
My quick grepping of CPLError CE_Debug shows only 4 call sites
```
frmts/rda/rdadataset.cpp:                CPLError( CE_Debug, CPLE_AppDefined,
frmts/hdf4/hdf4imagedataset.cpp:        CPLError( CE_Debug, CPLE_AppDefined,
frmts/ecw/ecwdataset.cpp:        CPLError( CE_Debug, CPLE_AppDefined,
frmts/ecw/ecwdataset.cpp:        CPLError( CE_Debug, CPLE_AppDefined,
```
And those should probably changed to CPLDebug() call for the clarity. I don't think we need to make a semantic difference between CPLDebug() and CPLError(CE_Debug, )